recipes/vcvrack-sdk/1.1.6/conanfile.py

```diff
@@ -0,0 +1,187 @@
+import os, stat, json, re, subprocess
+from conans import ConanFile, CMake, tools
+from conans.tools import os_info, SystemPackageTool
+from conans.errors import ConanInvalidConfiguration
+from conans.model import Generator
+
+
+class VCVRackSDKConan(ConanFile):
+    name = "vcvrack-sdk"
+    version = "1.1.6"
+    license = "GPL-3.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://vcvrack.com"
+    description = "VCV Rack SDK for Rack plugin development."
+
+    settings = "os", "compiler", "arch"
+
+    _SDK_DIR = "Rack-SDK"
+
+    def configure(self):
+        if self._isVisualStudioBuild:
+            self.output.error("Rack SDK on Windows is only compatible with MinGW GCC toolchain!")
+            raise ConanInvalidConfiguration("VCV Rack SDK is not compatible with Visual Studio!")
+
+        if self.settings.arch != "x86_64":
+            raise ConanInvalidConfiguration("VCV Rack SDK only supports x86_64 platform!")
+
+    def requirements(self):
+        if self.settings.os == "Windows":
+            self.requires.add("msys2/20190524")
+
+    def system_requirements(self):
+        packages = ["git", "zip", "make", "cmake", "jq"]
+        update_installer = True
+
+        if self.settings.os == "Windows":
+            self.output.warn("manipulate script internal environment - add MSYS_BIN to PATH for using pacman tool")
+            del os.environ["CONAN_SYSREQUIRES_SUDO"]
+            os.environ["PATH"] += os.pathsep + self.env["MSYS_BIN"]
+            packages = ["zip", "mingw-w64-x86_64-jq", "mingw-w64-x86_64-libwinpthread"]
+            update_installer = False
+
+        if self.settings.os == "Macos":
+            packages += ["autoconf", "automake", "libtool"]
+
+        if self.settings.os == "Linux":
+            mesaPackage = { "ubuntu": "libglu1-mesa-dev",
+                            "debian": "libglu1-mesa-dev",
+                            "fedora": "mesa-libGLU-devel",
+                            "centos": "mesa-libGLU-devel",
+                            "arch": "mesa"
+                          }
+            packages += [mesaPackage[os_info.linux_distro]]
+
+        installer = SystemPackageTool()
+        for package in packages:
+            installer.install(package, update=update_installer)
```

Comment:
You can't install using SytemPackageTools, but you can check: https://github.com/conan-io/conan-center-index/blob/a504b23f1bd435639e92bd063253ed922577395b/recipes/backward-cpp/all/conanfile.py#L87

For historical reasons, SytemPackageTools doesn't offer a way to track the dependency versions and its profile, which caused many problems in the past.

Reply:
When this means in the end I can't install dependecies automatically then there is no point to add this recipe to the conan index. I just try to add it here, because I asked about 3 times on bintray that they link my conan remote with the main remote so my remote doesn't need to be added explicit to use the recipe. But there was no reaction at all to my request.
The point of this recipe is basically to avoid manual setup the development environment.

Reply:
In the long-term we will have an OpenGL and a Mesa package. You can add it then :)